rholang/src/main/k/rholang.k

```diff
@@ -1,132 +1,1039 @@
 module RHO-SYNTAX
 imports SUBSTITUTION
 
-// Syntax for processes
-syntax Proc ::=
-              // Empty process
-                "Nil"
+// We need processes, including variables
+syntax Proc ::= ProcNoVars
+              | Id
+
+// We need processes, excluding variables
+syntax ProcNoVars ::=
+              // Ground terms, including the empty process and expressions
+                Ground
               // Listen
-              | "for(" Name "<-" Name "){" Proc "}" [binder]
+              | "for" "(" Names "<-" Name ")" "{" Proc "}" [binder]
               // Persistent listen
-              | "for(" Name "<=" Name "){" Proc "}" [binder]
+              | "for" "(" Names "<=" Name ")" "{" Proc "}" [binder]
               | "contract" Name "(" Name ")" "=" "{" Proc "}" [binder]
               // Send
-              | Name "!(" Proc ")"
+              | Name "!(" Procs ")"
               // Evaluate
               | "*" Name
               // New
-              | "new" Id "in" "{" Proc "}" [binder]
+              | "new" Ids "in" "{" Proc "}" [binder]
               // Parallel
               | Proc "|" Proc [left]
-              // Misc
+
+// We put "Nil" and Exp in the same syntactic category, since they are
+// structurally equivalent as ground properties
```

Comment:
Ah, right. I need to rephrase that wording then. It is useful to have them grouped because of that top-level semantic equivalence but you're completely right that they are neither structurally nor semantically equivalent.